libraries/adaptive-expressions/src/builtinFunctions/rand.ts

```diff
@@ -19,22 +23,37 @@ export class Rand extends ExpressionEvaluator {
      * Initializes a new instance of the [Rand](xref:adaptive-expressions.Rand) class.
      */
     public constructor() {
-        super(ExpressionType.Rand, Rand.evaluator(), ReturnType.Number, FunctionUtils.validateBinaryNumber);
+        super(ExpressionType.Rand, Rand.evaluator, ReturnType.Number, FunctionUtils.validateBinaryNumber);
     }
 
-    /**
-     * @private
-     */
-    private static evaluator(): EvaluateExpressionDelegate {
-        return FunctionUtils.applyWithError((args: any[]): any => {
-            let error: string;
-            if (args[0] > args[1]) {
-                error = `Min value ${args[0]} cannot be greater than max value ${args[1]}.`;
-            }
+    private static evaluator(expression: Expression, state: MemoryInterface, options: Options): ValueWithError {
+        let result;
+        let minValue;
+        let maxValue;
+        let error: string;
+
+        ({ value: minValue, error } = expression.children[0].tryEvaluate(state, options));
```

Comment:
I think it reads cleaner to extract these child values before using them:

```typescript
const [maybeMinValue, maybeMaxValue] = expression.children;
...
... = maybeMinValue.tryEvaluate(state, options);
```

Reply:
Some type checking may be useful as well, though I'm not sure if that's the convention in `adaptive-expressions`.